libs/tracing/Tracing.mli

```diff
@@ -2,10 +2,41 @@
    See header of Tracing.ml for details
  *)
 
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+type analysis_flags = {
+  secrets_validators : bool;
+  historical_scan : bool;
+  allow_all_origins : bool;
+  deep_intra_file : bool;
+  deep_inter_file : bool;
+}
```

Comment:
Could we document what each of these is? Also nice if we say that deep_intra/inter are mutually exclusive or not.